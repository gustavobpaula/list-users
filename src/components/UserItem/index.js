import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  Name,
  Company,
  StyledLink,
} from './style';

export default function UserItem(
  {
    id,
    name,
    address,
    company,
  },
) {
  return (
    <Card>
      <StyledLink to={`/user/${id}`}>
        <Name>{name}</Name>
        <Company>{company.name} - {address.city}</Company>
      </StyledLink>
    </Card>
  );
}

UserItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.shape({
    city: PropTypes.string.isRequired,
  }).isRequired,
  company: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }).isRequired,
};
