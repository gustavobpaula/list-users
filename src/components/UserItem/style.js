import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Card = styled.article`
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 10px;
  cursor: pointer;
  margin: 5px;
  max-width: 300px;
  padding: 20px 30px;
  transition: background-color .3s ease;

  &:hover {
    background-color: #eee;
  }
`;

export const Name = styled.h3`
  color: #000;
  margin: 0 0 20px 0;
  text-decoration: none;
`;

export const Company = styled.p`
  color: #999;
  margin: 0;
  text-decoration: none;
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
`;
