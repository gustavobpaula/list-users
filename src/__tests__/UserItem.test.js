import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserItem from '../components/UserItem';

describe('Testing UserItem Component', () => {
  it('should render correctly', () => {
    const props = {
      id: 1,
      name: 'Lorem Ipsum',
      address: {
        city: 'Lorem Ipsum',
      },
      company: {
        name: 'Lorem Ipsum',
      },
    };

    const wrapper = shallow(
      <UserItem
        id={props.id}
        name={props.name}
        address={props.address}
        company={props.company}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
