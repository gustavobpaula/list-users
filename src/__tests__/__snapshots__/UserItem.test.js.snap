// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserItem Component should render correctly 1`] = `
<styled.article>
  <Styled(Link)
    to="/user/1"
  >
    <styled.h3>
      Lorem Ipsum
    </styled.h3>
    <styled.p>
      Lorem Ipsum
       | 
      Lorem Ipsum
    </styled.p>
  </Styled(Link)>
</styled.article>
`;
